packages/calcite-components/src/components/block/block.scss

```diff
@@ -150,18 +171,18 @@ calcite-handle {
 }
 
 .toggle-icon {
-  @apply text-color-3
-  transition-color
+  @apply transition-color
   self-center
   justify-self-end
   duration-150
   ease-in-out;
 
   margin-inline-end: var(--calcite-spacing-md);
+  color: var(--calcite-block-header-icon-color, var(--calcite-color-text-3));
 }
 
 .toggle:hover .toggle-icon {
-  @apply text-color-1;
+  color: var(--calcite-block-header-icon-color, var(--calcite-color-text-1));
```

Comment:
the actual design pattern is that the toggle gets the header title color on hover so this can be simplified to 

`color: var(--calcite-block-header-text-color, var(--calcite-color-text-1));`